src/ansys/pyensight/core/utils/dsg_server.py

```diff
@@ -830,17 +840,25 @@ def _handle_update_command(self, cmd: dynamic_scene_graph_pb2.SceneUpdateCommand
             name = "Texture update"
         self.log(f"{name} --------------------------")
 
-    def _finish_part(self) -> None:
+    def _finish_part(self) -> bool:
         """Complete the current part
 
         There is always a part being modified.  This method completes the current part, committing
         it to the handler.
+
+        Returns
+        -------
+        bool
+            True on success, False on failure
         """
+        ret = True
         try:
             self._callback_handler.finalize_part(self.part)
         except Exception as e:
             self.warn(f"Error encountered while finalizing part geometry: {str(e)}")
+            ret = False
         self._mesh_block_count += 1
+        return ret
 
     def _handle_part(self, part_cmd: Any) -> None:
         """Handle a DSG UPDATE_PART command

```

Comment:
In _handle_part() there's a _finish_part() call.  If there's a failure in the middle of handling parts, I would guess it will happen there.  Is there a code path to clean up the output at that point?  Does the code just error out of each _handle_part() call?

Reply:
While not ideal yet (we need the back channel on 'idle' to report failure modes), this should be ok.  It ensures that the dsg stream is fully processed, but it might report more warnings than ideal.